lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -4804,23 +4806,176 @@ LogicalResult StmtEmitter::emitVerifClockedAssertLike(
+  if (auto disable = op.getDisable())
+    return emitVerifClockedAssertLike(op, op.getProperty(), disable,
+                                      op.getClock(), op.getEdge(),
+                                      PPExtString("assert"));
+
+  // Try to identify a disable pattern in the property
+  Value disableVal, property;
+  bool disablePattern =
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::OrOp>(mBool(&disableVal), mProperty(&property))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::OrOp>(mProperty(&property), mBool(&disableVal))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<ltl::OrOp>(mProperty(&property), mBool(&disableVal))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<ltl::OrOp>(mBool(&disableVal), mProperty(&property)));
+
+  if (!disablePattern) {
+    mlir::OpBuilder builder(op.getContext());
+    mlir::Value f = builder.createOrFold<hw::ConstantOp>(
+        op.getLoc(), builder.getI1Type(), 0);
+
+    return emitVerifClockedAssertLike(op, op.getProperty(), f, op.getClock(),
+                                      op.getEdge(), PPExtString("assert"));
+  }
+  return emitVerifClockedAssertLike(op, property, disableVal, op.getClock(),
+                                    op.getEdge(), PPExtString("assert"));
 }
 
 LogicalResult StmtEmitter::visitVerif(verif::ClockedAssumeOp op) {
-  return emitVerifClockedAssertLike(op, op.getProperty(), op.getDisable(),
-                                    op.getClock(), op.getEdge(),
-                                    PPExtString("assume"));
+  // If the op has a disable, then emit it traditionally
+  if (auto disable = op.getDisable())
+    return emitVerifClockedAssertLike(op, op.getProperty(), disable,
+                                      op.getClock(), op.getEdge(),
+                                      PPExtString("assume"));
+
+  // Try to identify a disable pattern in the property
+  Value disableVal, property;
+  bool disablePattern =
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::OrOp>(mBool(&disableVal), mProperty(&property))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<comb::OrOp>(mProperty(&property), mBool(&disableVal))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<ltl::OrOp>(mProperty(&property), mBool(&disableVal))) ||
+      mlir::matchPattern(
+          op.getProperty(),
+          mlir::m_Op<ltl::OrOp>(mBool(&disableVal), mProperty(&property)));
+
+  if (!disablePattern) {
+    mlir::OpBuilder builder(op.getContext());
+    mlir::Value f = builder.createOrFold<hw::ConstantOp>(
+        op.getLoc(), builder.getI1Type(), 0);
+
+    return emitVerifClockedAssertLike(op, op.getProperty(), f, op.getClock(),
+                                      op.getEdge(), PPExtString("assume"));
+  }
```

Comment:
Please don't mutate IR in verilog emission. IR might be read from a different thread so it's not thread safe.

Reply:
Ah ok, I guess I can simply omit the disable all together, that would actually make a lot more sense.

Reply:
👍  You can pass in a null value where you currently have `f`, and then check in `emitVerifClockedAssertLike` whether the disable is non-null, and only then emit the `disable iff`.

Reply:
that's exactly what I just did lol